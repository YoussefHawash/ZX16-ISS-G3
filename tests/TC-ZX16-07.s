.text
.org 0x0000
    j main

.org 0x0020
main:
    li16 x6, 0x4000       # a0 = writable address
    li   x7, 20           # a1 = max input length

    ecall 0x001           # read string
    ecall 0x003           # print string
    ecall 0x00A           # exit