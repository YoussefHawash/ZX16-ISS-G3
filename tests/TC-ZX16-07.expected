J 32
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
LUI x6, 0x080
ADDI x6, 0x00
LI x7, 0x14
ECALL 1    # ReadString
ECALL 3    # PrintString
ECALL 10    # ProgramExit
ADD x0, x0
ADD x0, x0
ADD x0, x0